crates/rpc/rpc/src/engine/jwt_secret.rs

```diff
@@ -45,23 +45,38 @@ const JWT_SIGNATURE_ALGO: Algorithm = Algorithm::HS256;
 /// [Secret key - Engine API specs](https://github.com/ethereum/execution-apis/blob/main/src/engine/authentication.md#key-distribution)
 #[derive(Clone)]
 #[allow(missing_debug_implementations)]
-pub struct JwtSecret(String);
+pub struct JwtSecret([u8; 32]);
+
+impl TryInto<JwtSecret> for String {
```

Comment:
I'm not a fan of this `TryInto`, I'd rather have a `from_hex(AsRef<str>)`, which is more explicit.